spaceward/src/hooks/query/util.ts

```
@@ -0,0 +1,55 @@
+	const result: Expression = JSON.parse(expression, (k, v) => {
+		if (typeof v === "object") {
+			if (["arguments", "elements"].includes(k)) {
+				return Array.from(v);
+			}
+			const { Value, ...rest } = v;
+
+			if (!Value) {
+				return Object.entries(rest).reduce(
+					(acc, [k, v]) => {
+						return {
+							...acc,
```

Biome

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)